src/App.tsx

```diff
@@ -1,61 +1,131 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo, User } from './components/types';
+
+const findById = (id:number) => {
+  return usersFromServer.find(user => user.id === id) as User;
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [selectUser, setSelectUser] = useState(0);
+  const [hasSelectUserError, setHasSelectUserError] = useState(false);
+
+  const defaultTodos = todos.map(todo => {
+    return {
+      ...todo,
+      user: findById(todo.userId),
+    };
+  });
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setSelectUser(0);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectUser(+event.target.value);
+    setHasSelectUserError(false);
+  };
+
+  const todoId = (todosArr:Todo[]) => {
```

Comment:
`todosArr` is not good name...you can see that it is an array from type `:Todo[]`

please don't use words like 'arr', 'func', 'var'.. for names